src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -228,9 +289,24 @@ function Base.show(io::IO, h::HubbardRealSpace)
     println(io, ")")
 end
 
+"""
+    ==(H::HubbardRealSpace, G::HubbardRealSpace)
+
+Overload equality due to stored potential energy arrays.
+"""
+Base.:(==)(H::HubbardRealSpace, G::HubbardRealSpace) = all(map(p -> getproperty(H, p) == getproperty(G, p), propertynames(H)))
+
 starting_address(h::HubbardRealSpace) = h.address
-diagonal_element(h::HubbardRealSpace, address) = local_interaction(address, h.u)
-diagonal_element(h::HubbardRealSpace{1}, address) = local_interaction(address, h.u[1])
+function diagonal_element(h::HubbardRealSpace, address)
+    int = iszero(h.u) ? 0. : local_interaction(address, h.u)
+    pot = iszero(h.v) ? 0. : trap_potential(address, h.potential)
+    return int + pot
+end
+function diagonal_element(h::HubbardRealSpace{1}, address)
+    int = iszero(h.u[1]) ? 0. : local_interaction(address, h.u[1])
+    pot = iszero(h.v[1]) ? 0. : trap_potential(address, h.potential[1])
+    return int + pot
+end
```

Comment:
Why the special case for a single component?

Reply:
I belive `local_interaction` expects a number if a single-component address is used.